src/main/java/oof/command/RecurringCommand.java

```diff
@@ -0,0 +1,177 @@
+package oof.command;
+
+import oof.task.Deadline;
+import oof.task.Event;
+import oof.task.Task;
+import oof.task.Todo;
+import oof.Storage;
+import oof.TaskList;
+import oof.Ui;
+import oof.exception.OofException;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.InputMismatchException;
+
+/**
+ * Represents a command to select and recur a task.
+ */
+public class RecurringCommand extends Command {
+
+    private int index;
+    private int count;
+    private int frequency;
+    private static final int COUNT_MIN = 1;
+    private static final int COUNT_MAX = 10;
+    private static final int DAILY = 1;
+    private static final int WEEKLY = 2;
+    private static final int MONTHLY = 3;
+    private static final int YEARLY = 4;
+    private static final int WEEK = 7;
```

Comment:
`WEEK` doesn't explain to the reader anything. Is it the number of days in a week?